addons/docs/src/frameworks/web-components/custom-elements.ts

```diff
@@ -50,30 +50,58 @@ function mapData(data: TagItem[], category: string) {
   return (
     data &&
     data
-      .filter((item) => !!item)
+      .filter((item) => item && item.name)
       .reduce((acc, item) => {
         if (item.kind === 'method') return acc;
 
-        const type =
-          category === 'properties' ? { name: item.type?.text || item.type } : { name: 'void' };
-        acc[item.name] = {
-          name: item.name,
-          required: false,
-          description: item.description,
-          type,
-          table: {
-            category,
-            type: { summary: item.type?.text || item.type },
-            defaultValue: {
-              summary: item.default !== undefined ? item.default : item.defaultValue,
-            },
-          },
-        };
+        switch (category) {
+          case 'events':
+            mapEvent(item).forEach((argType) => {
+              acc[argType.name] = argType;
+            });
+            break;
+          default:
+            acc[item.name] = mapItem(item, category);
+            break;
+        }
+
         return acc;
       }, {} as ArgTypes)
   );
 }
 
+function mapItem(item: TagItem, category: string): ArgType {
+  const type =
+    category === 'properties' ? { name: item.type?.text || item.type } : { name: 'void' };
+
+  return {
+    name: item.name,
+    action: { name: item.name },
+    required: false,
+    description: item.description,
+    type,
+    table: {
+      category,
+      type: { summary: item.type?.text || item.type },
+      defaultValue: {
+        summary: item.default !== undefined ? item.default : item.defaultValue,
+      },
+    },
+  };
+}
+
+function mapEvent(item: TagItem): ArgType[] {
```

Comment:
Can you adda. couple unit tests for `mapEvent` to document the structure a little bit?

Reply:
Done!